dbms/src/Storages/KVStore/MultiRaft/PrehandleSnapshot.cpp

```diff
@@ -612,23 +678,32 @@ PrehandleResult KVStore::preHandleSSTsToDTFiles(
 
             // `split_keys` do not begin with 'z'.
             auto [split_keys, approx_bytes] = getSplitKey(log, this, new_region, sst_stream);
-
+            prehandling_trace.waitForSubtaskResources(region_id, split_keys.size() + 1);
```

Comment:
what about `prehandling_trace.waitForSubtaskResources(region_id, split_keys.size() + 1, ->getMaxParallelPrehandleSize());`
and remove `KVStore * kvstore` from `PreHandlingTrace`?

Reply:
In what class should we put `getMaxParallelPrehandleSize`?

Reply:
The function here is `KVStore::preHandleSSTsToDTFiles`, it can access to `KVStore::getMaxParallelPrehandleSize` by `this-> getMaxParallelPrehandleSize()`